src/source_range.rs

```diff
@@ -0,0 +1,112 @@
+use super::*;
+
+/// Serialize a stream of [pulldown-cmark-Events][Event] each with source string into a string-backed buffer.
+///
+/// Different from [`cmark_resume_with_options`], which always escape Markdown special characters like `#` or `[`,
+/// this function preserves special character escapes.
+/// That is, it only escapes a special character if it is escaped in the source.
```

Comment:
Can this be clearer? With the sentence preceding it, it feels like this says the opposite - it's supposed to preserve special character escapes?